neurodamus/utils/memory.py

```diff
@@ -271,3 +273,64 @@ def display_total(self):
         grand_total = pretty_printing_memory_mb(grand_total)
         logging.info("| {:<40s} | {:>12s} |".format("GRAND TOTAL", grand_total))
         logging.info("+{:-^57}+".format(""))
+
+    def total_memory_available():
+        """
+        Returns the total memory available in the system in MB
+        """
+        try:
+            virtual_memory = psutil.virtual_memory()
+            return virtual_memory.total / (1024 * 1024)  # Total available memory in MB
+        except Exception as e:
+            logging.error(f"Error: {e}")
+            return None
+
+    @run_only_rank0
+    def suggest_nodes(self, margin):
+        """
+        A function to calculate the suggested number of nodes to run the simulation
+        The function takes into account the fact that the memory overhead is
+        variable with the amount of ranks the simulation it's ran with.
+        One can also specify a custom margin to add to the memory usage.
+        """
+
+        try:
+            ranks_per_node = os.cpu_count()
+        except AttributeError:
+            ranks_per_node = multiprocessing.cpu_count()
+
+        full_overhead = self.base_memory * ranks_per_node
+
+        # initialize variable for iteration
+        est_nodes = 0
+        prev_est_nodes = None
+
+        while prev_est_nodes is None or est_nodes != prev_est_nodes:
+            prev_est_nodes = est_nodes
```

Comment:
Wouldnt it be simpler if we deducted the overhead from the total node amount (and also the margin) and divide ? Like that I guess we could avoid the loop

Reply:
I thought this approach would be a bit more accurate and in any case the loop doesn't take much time. It just makes sure that whatever is suggesting, it takes into consideration the escalating memory cost of the overhead.

If in any case you think it's better to take a simpler route, I can rework it as you suggested.

Reply:
My only concern is the risk of an infinite loop, but you know the math better. If this is guaranteed to finish within a few iterations Im ok with it.

Reply:
Technically if the (overhead_memory * number_of_ranks) < total_memory_on_node it should converge in every situation. But I guess that adding a timeout number of loops cannot hurt if you agree. 5-10 loops max just to be safe if you agree...